R/yaml.R

```diff
@@ -49,17 +49,19 @@ yaml_load = function(
   ))
   # the below simple parser is quite limited
   res = list()
-  r = '^( *)([^ ]+?):($|\\s+.*)'
+  r = '^(\\s*)([^[:blank:]]+?):($|\\s+.*)'
```

Comment:
I wish to keep it slightly simpler:

```suggestion
  r = '^(\\s*)([^ ]+?):($|\\s+.*)'
```

Reply:
If I understand this correctly, * should be greedy enough to eat all the whitespace and then we need something in there to match the rest of the keyword. I think even dot should work well. `r = '^(\\s*)(.+?):($|\\s+.*)' if you want to go even simpler.

Reply:
Right. Let's just use `.`. `[^ ]+` means the field name cannot contain spaces; `.` means the name can.